src/main/groovy/io/seqera/wave/redis/RedisFactory.groovy

```diff
@@ -37,19 +42,50 @@ import redis.clients.jedis.JedisPoolConfig
 @CompileStatic
 class RedisFactory {
 
+    final static private String ADDRESS_REGEX = '(rediss?://)?(?<host>[^:]+)(:(?<port>\\d+))?'
+
+    final static private Pattern ADDRESS_PATTERN = Pattern.compile(ADDRESS_REGEX)
+
     @Singleton
     JedisPool createRedisPool(
             @Value('${redis.uri}') String uri,
             @Value('${redis.pool.minIdle:0}') int minIdle,
             @Value('${redis.pool.maxIdle:10}') int maxIdle,
-            @Value('${redis.pool.maxTotal:50}') int maxTotal
+            @Value('${redis.pool.maxTotal:50}') int maxTotal,
+            @Value('${redis.client.timeout:5000}') int timeout,
+            @Nullable @Value('${redis.password}') String password
     ) {
         log.info "Using redis $uri as storage for rate limit - pool minIdle: ${minIdle}; maxIdle: ${maxIdle}; maxTotal: ${maxTotal}"
+        boolean ssl = uri.startsWith("rediss")
+        DefaultJedisClientConfig clientConfig = DefaultJedisClientConfig.builder()
+                .password(password)
```

Comment:
Is it safe if the password is `null`?

Reply:
I am making changes to use JedisURIHelper

Reply:
Not sure it's needed because I think that tries to lookup for the password in the Redis URI, instead we need to specify independently

Reply:
I have added apssword in config and if its null then check in URI

Reply:
password will be null if auth is not set, like currently we don't have that

should we make it compulsory?

Reply:
ok, does not need to be compulsory